src/getMachines.py

```diff
@@ -0,0 +1,51 @@
+import boto3
+from decEncoder import *
+
+"""Scan dynamo table"""
+dynamodb = boto3.resource('dynamodb', region_name = 'us-east-2')
+machineTable = dynamodb.Table('Machines')
+response = machineTable.scan()
+
+"""Lambda handler function for /buildings/{buildingId}/machines API call
+   Returns all the machines in a given building if no queries specified
+   If query (machineType and/or status) specified, filter machines matching those query parameters
+   If no buildingId or queries are specified, return all machines
+"""
+def getMachines(event, context):
+
+    buildingId = None
+    if event.get('pathParameters'):
+        buildingId = event.get('pathParameters').get('buildingId')
+	
+    machineType = None
+    status = None
+    if event.get('queryStringParameters'):
+	    machineType = event.get('queryStringParameters').get('machineType')
+	    status = event.get('queryStringParameters').get('status')
```

Comment:
It's probably best to build the ScanFilter (http://boto3.readthedocs.io/en/latest/reference/services/dynamodb.html#DynamoDB.Table.scan) here. We would start with an empty dictionary, if machineType is defined it sets `scanFilter['machineType'] = {'AttributeValueList': [event.get('queryStringParameters').get('machineType')], 'ComparisonOperator': 'EQ'}`, and similar for status (note: that code may or may not work. I didn't try it.).